File: swarm2/team-agent/frontend/src/services/trust.service.ts
import apiClient from './api.client'
import type { TrustMetrics, TrustEvent, TrustHistory } from '@/types/trust.types'

export class TrustService {
  async getAllAgents(): Promise<TrustMetrics[]> {
    const response = await apiClient.get('/trust/agents')
    return response.data
  }

  async getAgentDetails(agentId: string): Promise<TrustMetrics> {
    const response = await apiClient.get(`/trust/agent/${agentId}`)
    return response.data
  }

  async getAgentHistory(agentId: string): Promise<TrustHistory> {
    const response = await apiClient.get(`/trust/agent/${agentId}/history`)
    return response.data
  }

  async getAgentEvents(agentId: string): Promise<TrustEvent[]> {
    const response = await apiClient.get(`/trust/agent/${agentId}/events`)
    return response.data
  }

  async recordEvent(agentId: string, eventData: Partial<TrustEvent>): Promise<void> {
    await apiClient.post(`/trust/agent/${agentId}/event`, eventData)
  }
}

export default new TrustService()

File: swarm2/team-agent/backend/app/services/trust_service.py
"""
Trust Service - Bridges Flask API to AgentReputationTracker.
"""
import sys
import os

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../../..')))

from typing import Dict, Any, List, Optional


class TrustService:
    """
    Service layer for trust management.
    Bridges Flask API to AgentReputationTracker.
    """

    def __init__(self):
        # TODO: Initialize trust tracker when ready
        # from swarms.team_agent.crypto.trust import AgentReputationTracker
        # self.tracker = AgentReputationTracker()
        self.agents = {}

    def get_all_agents(self) -> List[Dict[str, Any]]:
        """
        Get all agents with trust metrics.

        Returns:
            List of agent metrics
        """
        # TODO: Get from tracker.get_agent_metrics()
        return list(self.agents.values())

    def get_agent_details(self, agent_id: str) -> Optional[Dict[str, Any]]:
        """Get detailed trust metrics for an agent."""
        # TODO: Get from tracker.get_agent_metrics(agent_id)
        return self.agents.get(agent_id)

    def get_agent_history(self, agent_id: str) -> Dict[str, Any]:
        """Get trust score history for an agent."""
        # TODO: Get from tracker.get_trust_history(agent_id)
        return {
            'agent_id': agent_id,
            'data_points': []
        }

    def get_agent_events(self, agent_id: str) -> List[Dict[str, Any]]:
        """Get trust events for an agent."""
        # TODO: Get from tracker event log
        return []

    def record_event(self, agent_id: str, event_data: Dict[str, Any]) -> None:
        """Record a new trust event."""
        # TODO: Call tracker.record_event()
        pass


# Singleton instance
trust_service = TrustService()
